### PR TITLE
Added generic interface fortran wrappers with tests

### PR DESCRIPTION
**Pull request by brtnfld**
Added fortran wrappers for H5Pset_zfp_*_cdata.
    Added tests for new wrappers.
    Cleaned-up Fortran make check output.
    Updated Fortran wrapper docs

### CODE REVIEW

**Comment by markcmiller86** [comment]
Nah. I wouldn't sweat it. This is fine as is. Thanks for doing it!

**Comment by brtnfld** [comment]
The last two commits fix issues on skybridge (actually the Make fix is for everywhere).

**Comment by brtnfld** [comment]
removed findstring and used grep instead to detect the form of  HDF5_HOME.

**Comment by brtnfld** [comment]
@markcmiller86 The squashed commit message:

Added fortran wrappers for H5Pset_zfp_*_cdata.
Added tests and updated docs for new wrappers.
Cleaned-up Fortran make check output and fixed the detection of the HDF5_HOME format.
Fixed BOZ compilation error when it cannot appear in an old-style initialization.

File: docs/interfaces.rst [hunk 1]
```diff
@@ -147,11 +144,10 @@ is whether the application calls ``H5Z_zfp_initialize()`` or not.
 Fortran Interface
 -----------------
 
-A Fortran interface based on the properties interface, described above,
+A Fortran interface based on the properties interface and generic interface described above
```
Comment (markcmiller86) [inline docs/interfaces.rst:147]: Can I recommend a minor revision...

```
Fortran equivalents for both the properties and generic interfaces, described above,
```

Comment (brtnfld) [reply docs/interfaces.rst]: done.

**Pull request merged**